File: 2687-lexicographically-smallest-beautiful-string/lexicographically-smallest-beautiful-string.cpp
class Solution {
public:
    string smallestBeautifulString(string s, int k) {
        int n = s.size();
        string inp = s;
        bool bad = true;
        for(int i = n-1;i>=0;i--){
            bool isPossible = true;
            for(int j = i;j<n;j++){ /// will run 2-4 times at most
                char atLeast = 'a';
                if(j == i){
                    atLeast = s[i]+1;
                }
                bool found = false;
                for(int c = atLeast;c<=('a'+k-1);c++){ // will run 2-4 times at most
                    if((j == 0 || s[j-1]!=c) && (j<=1 || s[j-2]!=c)){
                        s[j] = c;
                        found = true;
                        break;
                    }
                }
                if(!found) {isPossible = false; break;}
            }
            if(isPossible) {bad = false; break;}
            else{
                for(int l = i;l<min(n,i+2);l++) s[i] = inp[i];
            }
        }
        if(bad) return "";
        return s;
    }
};

File: 2687-lexicographically-smallest-beautiful-string/lexicographically-smallest-beautiful-string.java
class Solution {
    public String smallestBeautifulString(String S, int K) {
        int N=S.length();
         char[] s = S.toCharArray();
        for(int i = N -1; i >=0; i--){
            for(char ch = (char)(s[i] +1); ch < 'a' + K; ch++){
                if(isValid(s, i, ch)){
                    s[i] = ch;
                    boolean possible = true;
                    for(int j = i +1; j < N; j++){
                        boolean found = false;
                        for(char c = 'a'; c < 'a' + K; c++){
                            if(isValid(s, j, c)){
                                s[j] = c;
                                found = true;
                                break;
                            }
                        }
                        if(!found){
                            possible = false;
                            break;
                        }
                    }
                    if(possible){
                        return new String(s);
                    }
                }
            }
        }
        return "";
    }
     public static boolean isValid(char[] s, int pos, char ch){
        if(pos >=1 && s[pos -1] == ch){
            return false;
        }
        if(pos >=2 && s[pos -2] == ch){
            return false;
        }
        return true;
    }
}